### ISSUE TITLE
ansible-lint fails to load role templates while ansible succeeds

### ISSUE THREAD

**Issue by ssbarnea** [ansible/ansible-lint#375]
# Issue Type
- Bug report

# Ansible and Ansible Lint details

```
ansible 2.5.9
ansible-lint 3.5.0
```

# Desired Behaviour

Avoinding errors about missing templates.

# Actual Behaviour (Bug report only)

```
 [WARNING]: Unable to find 'foo.xml.j2' in expected paths (use
-vvvvv to see paths)
```

**Comment by D3DeFi** [comment]
Maybe unrelated, but I am hitting the same problem with ansible `lookup('file', 'data.txt')` plugin in `block`:

```yaml
---
# tasks file for block-test role
- name: test1 will pass lint
  copy:
    content: "{{ lookup('file', 'data.txt') }}"
    dest: /tmp/test1.txt

- name: block here
  block:
    - name: test2 will fail lint
      copy:
        content: "{{ lookup('file', 'data.txt') }}"
        dest: /tmp/test2.txt
```

The 2nd one will present:
```
 [WARNING]: Unable to find 'data.txt' in expected paths (use -vvvvv to see paths)
```

I did this to create new role:
```bash
~$ molecule init role -r block-test -d docker
~$ cd block-test
~$ mkdir files; echo 'some data' > files/data.txt
```

**Comment by MaciejKucia** [comment]
The same issue

```
- block:
    - name: Setup sshd Banner message
      become: True
      copy:
        content: "{{lookup('file', 'files/banner')}}\n"
```

```
[WARNING]: Unable to find 'files/banner' in expected paths (use -vvvvv to see paths)
```

The other funny thing is that adding verbosity to _ansible-lint_ does not show any paths.

**Comment by juanvalino** [comment]
Same problem here +1

**Comment by ghislainbourgeois** [comment]
I can reproduce this issue with Ansible 2.8 and ansible-lint 4.2.0

**Comment by webknjaz** [comment]
This is probably solved.

**Comment by ppa-arenadata** [comment]
ansible 2.9.4
ansible-lint 4.2.0

`- block:
  - name: mytask
    tags: skip_ansible_lint  # dont hide warning
    uri:
      url: http://myserver
      method: POST
      body: "{{ lookup('template','mytemplate.json') }}"

Get: [WARNING]: Unable to find 'Arenadata_DB_system_cluster.json' in expected paths (use -vvvvv to see paths)

**Comment by alanbchristie** [comment]
*probably solved* isn't really a good reason to close an issue, especially as it's relatively easy to reproduce. I see the same issue with Ansible `2.9.5` and ansible-lint `4.2.0`. Granted, it's just a warning and the exit-code of the lint process is zero, but advice on how to suppress it would be appreciated.

```
[WARNING]: Unable to find 'secret-django.yaml.j2' in expected paths (use -vvvvv
to see paths)
```

**Comment by dwjvaughan** [comment]
Same issue here 3.5.1 (python 2.7)

**Comment by webknjaz** [comment]
@dwjvaughan that version is outdated and so is unsupported.

If anybody wants this resurrected, please submit a PR with pytest-based regression tests that catch this behavior as per https://pganssle-talks.github.io/xfail-lightning.

**Comment by webknjaz** [comment]
Also, "This is probably solved." was referring to master. So try reproducing there before making any further assumptions.

**Comment by ssbarnea** [comment]
At least in this case, I fully agree with @webknjaz. The linter is a 100%-community effort and is already hard to provide support for current version. I really like his idea of proposing xfail tests, as we can merge them right away as confirmation of the bug and prioritize work on them.

**Comment by dwjvaughan** [comment]
Appreciate the update everyone, I was hoping there was a work-around by _telling_ ansible-lint that it should look in the templates folder or something similar. It works if I hard-code the location of the template (but that's not very portable).

Unfortunately I'm stuck using python 2.7 so 3.5.1 is the latest I can go.

**Comment by webknjaz** [comment]
> so 3.5.1 is the latest I can go.

v4.2.0 still supports Python 2.7, newer won't.

**Comment by RemiCardona** [comment]
I can reproduce this issue with the current git master (`ansible-lint 4.3.0a2.dev32+gdbcdae2`) as well as the last release. I haven't tried building a test case but I _think_ that's going to be hard to do as I use `lookup` in a lot of different places and only 2 of the last I recently added fail.

Here's what my task looks like

```yaml
  - name: Copy Vault client configuration files
    copy:
      dest: "{{ adbv_configuration_path }}/vault-client/ca-int.key+cert.pem"
      content: |
        {{ lookup('file', 'ca-int.key.pem') }}
        {{ lookup('file', 'ca-int.cert.pem') }}
        {{ lookup('file', 'ca.cert.pem') }}
```

ansible-lint only complains about the first line and not the other 2, while the files are in the exact same location. And funnily enough, if I reorder the `content` block, it _still_ complains about the first line and not the other two.

If I add a static string as the first line of the `content` block, ansible-lint still complains about the first lookup (which is now on the second line).

And lastly, adding `skip_ansible_lint` does not work to silence the warning. So apart from doing a 3-line template to replace the copy+content, I'm running out of ideas. Cheers

**Comment by juanvalino** [comment]
Same problem here

**Comment by ssbarnea** [comment]
@juanvalino I do not want to disappoint you but is very unlikely someone will fix unless is one of the affected users, like you. We welcome PRs and I think we review them quite fast.
